src/devices/Hmc5883l/Hmc5883l.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Numerics;
+
+namespace Iot.Device.Hmc5883l
+{
+    /// <summary>
+    /// 3-Axis Digital Compass HMC5883L
+    /// </summary>
+    public class Hmc5883l : IDisposable
+    {
+        /// <summary>
+        /// HMC5883L I2C Address
+        /// </summary>
+        public const byte I2cAddress = 0x1E;
+
+        private I2cDevice _sensor;
+
+        private readonly byte _measuringMode;
+        private readonly byte _outputRate;
+        private readonly byte _gain;
+
+        /// <summary>
+        /// HMC5883L Direction Angle
+        /// </summary>
+        public double DirectionAngle => RawToDirectionAngle(ReadRaw());
+
+        /// <summary>
+        /// Initialize a new HMC5883L device connected through I2C
```

Comment:
Question: Does this binding support a different communication protocol as well? Or is I2c the only one?

Reply:
Only I2C. This passage was you write to me last time. I copied it directly.😅